source/code/plugin/KubernetesApiClient.rb

```diff
@@ -357,6 +374,14 @@ def getContainerResourceRequestsAndLimits(metricJSON, metricCategory, metricName
             podUid = pod["metadata"]["uid"]
           end
 
+          # For ARO, skip the pods scheduled on to master or infra nodes to ingest
+          if isAROCluster && !pod["spec"]["nodeName"].nil? &&
```

Comment:
wrap this is a begin exception end block, or check for pod["spec"].nil? as well.